example_test.go

```diff
@@ -314,3 +314,58 @@ func Example_customCommand() {
 	fmt.Printf("%q %s", v, err)
 	// Output: "" redis: nil
 }
+
+func ExampleScanIterator() {
+	iter := client.ScanIterator(nil)
+
+	for iter.Next() {
+		fmt.Println(iter.Value())
```

Comment:
The trouble is that it's `Key` for `SCAN` and `Field` for all the others. I rename it to `Val` to keep it consistent with `Cmd`